homepage/src/App.js

```diff
@@ -1,14 +1,28 @@
-import { useState } from 'react';
+import { useEffect, useState} from 'react';
 import Home from './components/Home';
 import LibraryComponents from './components/LibraryComponents';
+import ClipLoader from 'react-spinners/ClipLoader';
+import {fetchContentfulAPI} from './imageRendering';
 
 function App() {
   const [isHome, setIsHome] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+  const [imageURLS, setImageURLS] = useState([]);
+
+  // make the api call
+  useEffect(() => {
+    fetchContentfulAPI(imageURLS, setImageURLS, setIsLoading);
```

Comment:
For this part, I thought if you leave the array for the useEffect blank, then the callback function will only get called when the website first renders and will not repeat after. Initially, I left the array blank, but I think you filled it in after since it was giving it a warning? If leaving the array blank isn't the solution, lmk and I'll look into it. The function should be called once only on the first renderation of the website.

Reply:
I thinking leaving the array blank is not the way to go, but this is only because of the warning. I'd just go ahead and put in a check like `if (imageURLS == undefined || imageURLS.length == 0)` as it seems to be much more straightforward.